packages/axelar-wasm-std/src/voting.rs

```diff
@@ -193,22 +195,26 @@ impl WeightedPoll {
             quorum: snapshot.quorum,
             expires_at: expiry,
             poll_size: poll_size as u64,
-            votes: vec![Uint256::zero(); poll_size],
+            tallies: vec![Uint256::zero(); poll_size],
             status: PollStatus::InProgress,
             participation,
         }
     }
 }
 
 impl Poll for WeightedPoll {
-    fn tally(&mut self, block_height: u64) -> Result<PollResult, Error> {
+    fn complete(&mut self, block_height: u64) -> Result<PollResult, Error> {
```

Comment:
similar to my comment on cast_vote, I'd prefer to minimize side effects. This function sets the poll to finished (why the discrepancy in naming for complete <-> finished?) but mostly it computes the poll result. And then presumably the poll gets stored somewhere outside this context. I suggest splitting this of:
```
let completed_poll = poll.complete(...)?;
store.save(completed_poll);
return cumulate_poll_result(completed_poll);
```